app/src/main/java/com/example/weather/notification/DailyWorker.kt

```diff
@@ -0,0 +1,146 @@
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.singleOrNull
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import java.io.IOException
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class DailyWorker(context: Context, workerParameters: WorkerParameters) :
+    Worker(context, workerParameters), KoinComponent {
+
+    @SuppressLint("MissingPermission")
+    override fun doWork(): Result {
+        CoroutineScope(Dispatchers.Default).launch {
+            try {
+                val repository: WeatherRepository by inject()
+
+                val fusedLocationClient =
+                    LocationServices.getFusedLocationProviderClient(applicationContext)
+
+                val location = suspendCoroutine { continuation ->
+                    fusedLocationClient.lastLocation.addOnSuccessListener { location: Location? ->
+                        continuation.resume(location)
+                    }.addOnFailureListener { _ ->
+                        continuation.resume(null)
+                    }
+                }
+
+                if (location != null) {
+                    fetchWeather(repository, location)
+                }
+            } catch (e: IOException) {
+                Log.e("DailyWorker", "Exception occurred: $e")
+            }
+        }
+        return Result.success()
+    }
+
+    private suspend fun fetchWeather(
+        repository: WeatherRepository,
+        currentLocation: Location?
+    ) {
+        currentLocation?.let { currentLocation ->
+            CoroutineScope(Dispatchers.Default).launch {
+                val currentDeferred = async {
+                    repository.fetchWeatherForecastCurrent(
+                        currentLocation.latitude,
+                        currentLocation.longitude
+                    )
+                }
+                val data = currentDeferred.await().singleOrNull()
+
+                val location = data?.getLocation()
+                val weatherDescription = data?.weatherCurrent?.weatherDescription
+                val temperature = data?.weatherCurrent?.temperature?.kelvinToCelsius().toString()
+                val windSpeed = data?.weatherCurrent?.windSpeed?.mpsToKmph().toString()
+                val dateTime = data?.weatherCurrent?.dateTime?.unixTimestampToTimeString()
+                val notificationTitle = "$location          $dateTime"
+                val notificationContent =
+                    "$weatherDescription        ${
+                    applicationContext.getString(R.string.temperature)
+                    } $temperature℃        ${
+                    applicationContext.getString(R.string.wind_speed)
+                    } $windSpeed km/h"
+                showNotification(notificationTitle, notificationContent)
+            }
+        }
+    }
+
+    @SuppressLint("MissingPermission")
```

Comment:
```suggestion
    // FIXME Update later
    @SuppressLint("MissingPermission")
```
Ca này gấp gáp nên anh merged nhé, em add tag này vào sau phải fix không chơi như này

Reply:
@daolq-2712 dạ lát chiều em sẽ fix luôn ạ.